hydroflow_lang/src/graph/ops/join_multiset.rs

```diff
@@ -0,0 +1,69 @@
+use syn::{parse_quote, parse_quote_spanned};
+
+use super::{
+    FlowProperties, FlowPropertyVal, OperatorCategory, OperatorConstraints, WriteContextArgs,
+    RANGE_0, RANGE_1,
+};
+use crate::graph::{OpInstGenerics, OperatorInstance};
+
+/// > 2 input streams of type <(K, V1)> and <(K, V2)>, 1 output stream of type <(K, (V1, V2))>
+///
+/// This operator is equivalent to `join` except that the LHS and RHS are collected into multisets rather than sets before joining.
+///
+/// For example:
+/// ```hydroflow
+/// lhs = source_iter([("a", 0), ("a", 0)]) -> tee();
+/// rhs = source_iter([("a", 0)]) -> tee();
```

Comment:
I'd encourage you to make the RHS distinguishable to avoid confusion in the example.
rhs = source_iter(["a", "hydro")]) -> tee();

Reply:
@zzlk please fix the assert below the code as well